compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -499,7 +490,24 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
                             None => {}
                         }
                         // Return alloc mutability
-                        if is_mut { Mutability::Mut } else { Mutability::Not }
+                        if let DefKind::Static { mt, nested } = self.ecx.tcx.def_kind(did) {
+                            if nested {
+                                mt
+                            } else if !self
+                                .ecx
+                                .tcx
+                                .type_of(did)
```

Comment:
What would happen if we did `type_of` for a nested static? Would it ICE or return junk?